components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/DefaultContentValidator.java

```diff
@@ -55,38 +59,81 @@ public DefaultContentValidator(final MimeSupport mimeSupport) {
   @Override
   public String determineContentType(boolean strictContentTypeValidation,
                                      Supplier<InputStream> contentSupplier,
+                                     @Nullable MimeRulesSource mimeRulesSource,
                                      @Nullable String contentName,
                                      @Nullable String declaredContentType) throws IOException
   {
     checkNotNull(contentSupplier);
-    String contentType = declaredContentType;
 
-    if (contentType == null) {
-      log.trace("Content {} has no declared content type.", contentName);
-      try (InputStream is = contentSupplier.get()) {
-        contentType = mimeSupport.detectMimeType(is, contentName);
+    final List<String> contentDetectedMimeTypes = new ArrayList<>();
```

Comment:
If you use `Set` instead of `List` here then you don't need to convert it to `Set` later on (can use `LinkedHashSet` to preserve insertion order)